packages/legend-studio/src/stores/editor-state/element-editor-state/mapping/MappingTestState.ts

```diff
@@ -624,35 +618,22 @@ export class MappingTestState {
     }
     const startTime = Date.now();
     try {
-      const runtime = this.inputDataState.runtime;
       this.isRunningTest = true;
       const result =
-        (yield this.editorStore.graphManagerState.graphManager.executeMapping(
+        (yield this.editorStore.graphManagerState.graphManager.executeMappingTest(
           this.editorStore.graphManagerState.graph,
           this.mappingEditorState.mapping,
-          this.test.query,
-          runtime,
+          this.queryState.test.name,
           PureClientVersion.VX_X_X,
           true,
-        )) as ExecutionResult;
+        )) as MappingTestResult;
       this.testExecutionResultText = losslessStringify(
-        extractExecutionResultValues(result),
+        result.actual,
         undefined,
         TAB_SIZE,
       );
-      let assertionMatched = false;
-      if (
-        this.assertionState instanceof MappingTestExpectedOutputAssertionState
-      ) {
-        // TODO: this logic should probably be better handled in by engine mapping test runner
-        assertionMatched =
-          hashObject(extractExecutionResultValues(result)) ===
-          hashObject(losslessParse(this.assertionState.expectedResult));
-      } else {
-        throw new UnsupportedOperationError();
-      }
       this.setResult(
-        assertionMatched ? TEST_RESULT.PASSED : TEST_RESULT.FAILED,
+        result.result === 'SUCCESS' ? TEST_RESULT.PASSED : TEST_RESULT.FAILED,
```

Comment:
Let's use the correct enum and not the string `SUCCESS`

Reply:
We get this using legend-engine, so will make the change there

Reply:
We need a change in engine for this? It is already an Enum in engine. should be SUCESS, FAILURE, ERROR